source/extensions/filters/http/jwt_authn/filter_factory.cc

```diff
@@ -45,9 +45,22 @@ FilterFactory::createFilterFactoryFromProtoTyped(const JwtAuthentication& proto_
                                                  Server::Configuration::FactoryContext& context) {
   validateJwtConfig(proto_config);
   auto filter_config = std::make_shared<FilterConfig>(proto_config, prefix, context);
-  return [filter_config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+
+  auto factory = [filter_config](const std::vector<std::string>& audiences,
+                                 const absl::optional<std::string>& issuer,
+                                 std::vector<JwtLocationConstPtr>&& tokens) -> AuthenticatorPtr {
+    return Authenticator::create(audiences, issuer, std::move(tokens), filter_config);
```

Comment:
Only think you need is filter_config for this lambda function.   I prefer to pass filter_config to where ever places this "factory" is passing to.   and call Authenticator::create directly.   Such callback making code harder to read.

Reply:
I want the ability to inject mock authenticators in my verifier test. Should I pass in an utility class to create authenticators instead? Authenticator::create is static an not possbile to mock.

Reply:
I see.  it is OK then